Knp/Console/Application.php

```diff
@@ -4,30 +4,60 @@
 
 use Symfony\Component\Console\Application as BaseApplication;
 use Silex\Application as SilexApplication;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
 
+/**
+ * Silex console application class.
+ */
 class Application extends BaseApplication
 {
     private $silexApplication;
-
     private $projectDirectory;
 
+    /**
+     * Application constructor.
+     *
+     * @param SilexApplication $application      The Silex application
+     * @param string           $projectDirectory The project root directory path
+     * @param string           $name             The name of the application
+     * @param string           $version          The version of the application
+     */
     public function __construct(SilexApplication $application, $projectDirectory, $name = 'UNKNOWN', $version = 'UNKNOWN')
     {
         parent::__construct($name, $version);
 
         $this->silexApplication = $application;
         $this->projectDirectory = $projectDirectory;
-
-        $application->boot();
     }
 
+    /**
+     * Gets the Silex application.
+     *
+     * @return SilexApplication
+     */
     public function getSilexApplication()
     {
         return $this->silexApplication;
     }
 
+    /**
+     * Gets the project root path.
+     *
+     * @return string
+     */
     public function getProjectDirectory()
     {
         return $this->projectDirectory;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function doRun(InputInterface $input = null, OutputInterface $output = null)
+    {
```

Comment:
Not here. The parameters are optional in `run()` but not in `doRun()`. Good catch.